WooCommerce/WooCommerceTests/ViewRelated/Custom Fields/CustomFieldsListViewModelTests.swift

```diff
@@ -182,4 +196,61 @@ final class CustomFieldsListViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.combinedList.last?.value, "NewValue")
         XCTAssertNil(viewModel.combinedList.last?.fieldId)
     }
+
+    func test_given_savingSucceeds_when_saveChangesCalled_then_changesAreSaved() async {
+        // Given: successfully saving the changes
+        let newField = MetaData(metadataID: 10, key: "NewKey", value: "NewValue")
+        stores.whenReceivingAction(ofType: MetaDataAction.self) { [self] action in
+            switch action {
+                case let .updateMetaData(_, _, _, _, onCompletion):
+                    onCompletion(.success(originalMetadata + [newField]))
+            }
+        }
+
+        // When: Saving the changes
+        viewModel.saveField(key: newField.key, value: newField.value, fieldId: nil)
+        await viewModel.saveChanges()
+
+        // Then: The changes should be saved
+        XCTAssertEqual(viewModel.combinedList.count, originalFields.count + 1)
+        XCTAssertEqual(viewModel.combinedList.last?.key, newField.key)
+        XCTAssertEqual(viewModel.combinedList.last?.value, newField.value)
+        XCTAssertEqual(viewModel.combinedList.last?.fieldId, newField.metadataID)
```

Comment:
Nit: this can be simplified as:

```suggestion
        XCTAssertEqual(viewModel.combinedList, originalFields + [newField])
```

Reply:
It won't work like this simply, we need to update `CustomFieldUI` to be `Equatable` first, and then map from `originalFields` to a `[CustomFieldUI]` and also `[newField]` has to be mapped.

So I think what I have now is simpler, WDYT?